src/test/scala/org/broadinstitute/dsde/workbench/sam/api/ResourceRoutesV2Spec.scala

```diff
@@ -610,6 +610,118 @@ class ResourceRoutesV2Spec extends AnyFlatSpec with Matchers with TestSupport wi
+      status shouldEqual StatusCodes.NoContent
+    }
+  }
+
+  it should "403 removing without permission" in {
+    // differs from happy case in that owner role does not have alter_policies
+    val resourceType = ResourceType(ResourceTypeName("rt"), Set(SamResourceActionPatterns.alterPolicies, SamResourceActionPatterns.sharePolicy), Set(ResourceRole(ResourceRoleName("owner"), Set(SamResourceActions.sharePolicy(AccessPolicyName("splat"))))), ResourceRoleName("owner"))
+    val samRoutes = TestSamRoutes(Map(resourceType.name -> resourceType))
+    runAndWait(samRoutes.resourceService.createResource(resourceType, ResourceId("foo"), defaultUserInfo, samRequestContext))
+    runAndWait(samRoutes.resourceService.createResource(resourceType, ResourceId("bar"), defaultUserInfo, samRequestContext))
+
+    runAndWait(samRoutes.resourceService.addSubjectToPolicy(
+      FullyQualifiedPolicyId(FullyQualifiedResourceId(resourceType.name, ResourceId("foo")),
+        AccessPolicyName(resourceType.ownerRoleName.value)),
+      FullyQualifiedPolicyId(FullyQualifiedResourceId(resourceType.name, ResourceId("bar")),
+        AccessPolicyName("reader")),
+      samRequestContext
+    ))
+
+    Delete(s"/api/resources/v2/${resourceType.name}/foo/policies/${resourceType.ownerRoleName}/memberPolicies/${resourceType.name}/bar/reader") ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.Forbidden
+    }
+  }
+
+  it should "404 removing without any access" in {
+    // differs from happy case in that testUser creates resource, not defaultUser which calls the PUT
+    val resourceType = ResourceType(ResourceTypeName("rt"), Set(SamResourceActionPatterns.alterPolicies, ResourceActionPattern("can_compute", "", false)), Set(ResourceRole(ResourceRoleName("owner"), Set(ResourceAction("can_compute")))), ResourceRoleName("owner"))
+    val samRoutes = TestSamRoutes(Map(resourceType.name -> resourceType))
+    val testUser = Generator.genWorkbenchUserGoogle.sample.get
+
+    runAndWait(samRoutes.userService.createUser(testUser, samRequestContext))
+
+    runAndWait(samRoutes.resourceService.createResource(resourceType, ResourceId("foo"), testUser, samRequestContext))
+    runAndWait(samRoutes.resourceService.createResource(resourceType, ResourceId("bar"), testUser, samRequestContext))
+
+    runAndWait(samRoutes.resourceService.addSubjectToPolicy(
+      FullyQualifiedPolicyId(FullyQualifiedResourceId(resourceType.name, ResourceId("foo")),
+        AccessPolicyName(resourceType.ownerRoleName.value)),
+      FullyQualifiedPolicyId(FullyQualifiedResourceId(resourceType.name, ResourceId("bar")),
+        AccessPolicyName("reader")),
+      samRequestContext
+    ))
+
+    Delete(s"/api/resources/v2/${resourceType.name}/foo/policies/${resourceType.ownerRoleName}/memberPolicies/${resourceType.name}/bar/reader") ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.NotFound
+    }
+  }
+
```

Comment:
assuming you add a runtime assertion that the user must have >= read on the memberPolicy to-be-added, add unit tests to verify that

also, how about a unit test to verify behavior of memberPolicies in the overwrite API?